### PR TITLE
Ark std rand

### PR DESCRIPTION
**Pull request by ValarDragon**
## Description

Moves ark-ff::UniformRand to be a re-export of ark-std's version. Deprecates ark-ff::test_rng().

closes: #113 
---

Before we can merge this PR, please make sure that all the following items have been
checked off. If any of the checklist items are not applicable, please leave them but
write a little note why.

- [x] Targeted PR against correct branch (main)
- [x] Linked to Github issue with discussion and accepted design OR have an explanation in the PR that describes this work.
- [x] Wrote unit tests - n/a
- [x] Updated relevant documentation in the code
- [x] Added a relevant changelog entry to the `Pending` section in `CHANGELOG.md`
- [x] Re-reviewed `Files changed` in the Github PR explorer

### CODE REVIEW

**Comment by ValarDragon** [comment]
That second commit shows that the deprecation is working as intended!

File: CHANGELOG.md [hunk 1]
```diff
@@ -11,12 +11,14 @@
...
 - #53 (ark-poly) Add `Zero` trait bound to `Polynomial`.
+- #96 (ark-ff) Make the `field_new` macro accept values in integer form, without requiring decomposition into limbs, and without requiring encoding in Montgomery form.
 - #106 (ark-ff, ark-ec) Add `Zeroize` trait bound to `Field, ProjectiveGroup, AffineGroup` traits.
 - #108 (ark-ff) Add `extension_degree()` method to `Field`.
 - #110 (ark-ec) Change the trait bound on the scalar for `mul`, from (essentially) `Into<BigInt>` to `AsRef<[u64]>`
 - #117 (ark-poly) Make the univariate `SparsePolynomial` implement `Polynomial`. Make this change
     by replacing `sparse_poly.evaluate(pt)` to `sparse_poly.evaluate(&pt)`.
-- #96 (ark-ff) Make the `field_new` macro accept values in integer form, without requiring decomposition into limbs, and without requiring encoding in Montgomery form.
+- #129 (ark-ff) Move `ark-ff::{UniformRand, test_rng}` to `ark-std::{UniformRand, test_rng}`.
+    Importing these from `ark-ff` is still possible, but deprecated and will be removed in the following release.
```
Comment (Pratyush) [inline CHANGELOG.md:21]:
```suggestion
- #129 (ark-ff) Move `ark_-_ff::{UniformRand, test_rng}` to `ark_std::{UniformRand, test_rng}`.
    Importing these from `ark-ff` is still possible, but is deprecated and will be removed in the following release.
```

**Comment by Pratyush** [comment]
Hmm unfortunately due to how we handle warnings, this change will cause downstream crates to fail to compile (because we `#[deny(deprecated)]`

**Comment by ValarDragon** [comment]
Hrmm, its only our repos that deny deprecated, perhaps other downstream users keep it at warn?
(Also perhaps we should change from deny(deprecated) to warn(deprecated) anyway?)

**Pull request merged**